src/libraries/System.Net.Http/src/System/Net/Http/Headers/ObjectCollection.cs

```diff
@@ -94,8 +94,9 @@ public void Clear()
         }
 
         public bool Contains(T item) =>
-            ReferenceEquals(item, _items) ||
-            (_size != 0 && _items is T[] items && Array.IndexOf(items, item, 0, _size) != -1);
+            _size == 0 || _items is null ? false :
```

Comment:
In what situation would _size != 0 but _items == null?  I'm wondering why the second check is needed (and worst case, if there's some corner case that could result in that, it would fall out of the subsequent type checks, both of which would be false).

Reply:
Just being defensive, but I guess at that point everything would be off.